PR 2 — Payments Creation + Callback Endpoints

tanishkgupta-18 opened the pull request:
Hi @Umang01-hash,

I am planning to build the following in this PR:

**What:**
- Design DB schema for payments table (id, amount, status, created_at, updated_at, etc.)
- Write migration for payments table
- Implement CreatePayment API: handler, service, store
- Implement PaymentCallback endpoint: update payment status
- Add Swagger docs for both endpoints
- Add unit tests for handler and service

**Timeline:**
- Day 1: Schema + migration + store
- Day 2: Service + handler + routes
- Day 3: Swagger update + unit tests

**Deliverables:**
- `/payments` POST endpoint
- `/payments/callback` POST endpoint
- Migration file
- Unit tests
- Updated Swagger docs

Please review this plan and let me know if you have any suggestions or changes before, I start implementing.

Thanks a lot for your guidance!

tanishkgupta-18 committed:
docs: add plan for PR 2

Umang01-hash commented:
@tanishkgupta-18 Looks good and fine to me. Please proceed ahead

tanishkgupta-18 committed:
add payments table migration

tanishkgupta-18 committed:
add routes

tanishkgupta-18 committed:
update swagger docs

tanishkgupta-18 committed:
add test

tanishkgupta-18 commented:
Hi @Umang01-hash,

I've completed the following:

- /payments POST endpoint

- /payments/callback POST endpoint

- Migration file for payments table

- Unit tests for the handler layer

- Swagger documentation updated

Could you please review the latest code and share any suggestions or improvements?

Thanks a lot for your guidance!

Umang01-hash commented on handler/payment_handler_test.go:
```diff
@@ -0,0 +1,140 @@
+			ctx.Request = gofrHttp.NewRequest(req)
+
+			got, err := h.CreatePayment(ctx)
+			if tt.name == "Invalid JSON" {
```
We should never have `if` conditions inside of a test method. If needed break down the method into more smaller methods with success and error cases separated out but don;t add these if else inside test methods.

Umang01-hash reviewed (changes requested):
- Why have we deleted the `configs/.env` file? We need it for configuring elements in GoFr and maybe other useful configs will also stay there.
- Let's add a `golangci-lint.yml` file (you can use the one in GoFr directory) and run the `golangci-lint run` command and resolve all the issues there.
- Let's add a postman collection with assertions to actually test the endpoints.

tanishkgupta-18 committed:
fix payment_handler_test file

tanishkgupta-18 committed:
fix the issues

tanishkgupta-18 committed:
add postman collection

tanishkgupta-18 commented:
Hi @Umang01-hash,
I've pushed the changes as per the feedback:

- Restored the `configs/.env` file

- Fixed all lint issues

- Added Postman collection with proper assertions

- Removed all if conditions from test

Please let me know if there’s anything else I should improve or revisit.

Thanks a lot for your guidance!

Umang01-hash commented on configs/.env:
```diff
@@ -0,0 +1,14 @@
+DB_DIALECT=mysql
+
+# Optional
+GOFR_TELEMETRY=false
```
We should enable it. It will eventually give us infor about performance, errors etc. You can set the  config to `TRACE_EXPORTER=gofr` for now.

Umang01-hash commented on handler/health.go:
```diff
@@ -8,6 +8,6 @@ func New() *Health {
 }
 
-func (h *Health) Health(_ *gofr.Context) (any, error) {
+func (*Health) Health(_ *gofr.Context) (any, error) {
```
One thought on my mind was since GoFr already provides two built-in health endpoints:

- `/.well-known/alive` - Simple UP status
- `/.well-known/health` - Comprehensive health status including dependencies

Unless there are specific requirements not mentioned that necessitate a custom health endpoint, I recommend **removing the custom health handler** and using GoFr's built-in health check mechanisms.

tanishkgupta-18 committed:
fix .env file

tanishkgupta-18 committed:
fix the health check

tanishkgupta-18 commented:
Hi @Umang01-hash,

I've pushed the changes based on your feedback:

- Updated the configs/.env file

- Removed the custom health handler and enabled GoFr’s inbuilt health check

Please let me know if there’s anything else I should improve or revisit.

Thanks a lot for your guidance!

Umang01-hash reviewed (commented):
- Please add test for store layer using mocks.
- Please add test for service layer using mocks.
-
